content/security.md

```diff
@@ -0,0 +1,380 @@
+
+#### Passing options into publications
+
+For certain applications, for example pagination, you'll want to pass options into the publication to control things like how many documents should be sent to the client. There are some extra considerations to keep in mind for this particular case.
+
+1. **Passing a limit**: In the case where you are passing the `limit` option of the query from the client, make sure to set a maximum limit. Otherwise, a malicious client could request too many documents at once, which could raise performance issues.
+2. **Passing in a filter**: If you want to pass fields to filter on because you don't want all of the data, for example in the case of a search query, make sure to intersect the fields passed from the client with the fields it is allowed to see. Otherwise, a client could query on secret fields it's not supposed to be able to access.
+3. **Passing in fields**: If you want the client to be able to decide which fields of the collection should be fetched, make sure to intersect that with the fields that client is allowed to see, so that you don't accidentally send secret data to the client.
+
+### Served files
+
+Publications are not the only place the client gets data from the server. The set of source code files and static assets that are served by your application server could also potentially contain sensitive data:
+
+1. Business logic an attacker could analyze to find weak points
+1. Secret algorithms that a competitor could steal
+1. Secret API keys
+
+While the client-side UI of your application is basically open source, every application will have some secret code on the server that you don't want to share with the world.
+
+Secret business logic in your app should be located in code that only runs on the server. This means it is in the `server/` directory of your app, in a package that is only included on the server, or in a file inside a package that was loaded only on the server.
+
+Keep in mind that code inside `if (Meteor.isServer)` blocks is still sent to the client, it is just not executed. So don't put any secret code in there.
+
+Secret API keys should never be stored in your source code at all, the next section will talk about how to handle them.
+
+### Securing API keys
+
+Every app will have some secret API keys or passwords:
+
+1. Your database password
+1. API keys for external APIs
+
+These should never be stored as part of your app's source code in version control; the appropriate place to put secret keys is in a _settings file_ or an _environment variable_.
+
+Most of your app settings should be in JSON files that you pass in when starting your app. You can start your app with a settings file by passing the `--settings` flag:
+
+```sh
+# Pass development settings when running your app locally
+meteor --settings development.json
+
+# Pass production settings when deploying your app
+meteor deploy myapp.com --settings production.json
```

Comment:
Should we say something about where to store `production.json`? What do we recommend?

1. Some people store in GH repo, which is generally seen to be a bad idea™
2. Others use dropbox or the like, which is marginally better, but still fundamentally as insecure
3. On some projects we've passed onetimesecret.com or lasspass links around but it's hard to keep in sync